design/decisions/why-license/index.md

```diff
@@ -0,0 +1,114 @@
+neither of those, which we discussion below as well.
+
+### Software license
+
+Overall the Open Source community have licenses that work along two
+strands, **permissive** and **copy-left**. Both of these 'allow software
+to be freely used, modified, and shared' (see more detail on the [Open
+Source Initiative](https://opensource.org/licenses) website). We will be
+using the OSI's definition of free software which among other things
+talk about the source code being available to download and read/study,
+to allow for derived works to be created and distributed without
+violating the given license.
+
+The **Copy-Left licenses** generally state that if a third party makes
+changes to the existing product, or incorporates the code alongside
+another code set, then the resulting software must also be available
+under the same license. This is the broadest definition of free
+software, as it forces any subsequent development to be shared for free
+(although most of those licenses state that you are allowed to modify
+code for own use, the copy-left license only comes into play if you make
+your modifications available outside your organisation). Examples of
+Copy-Left licenses are EUPL-1.2, GPL, and LGPL.
+
+The **Permissive licenses** gives you all the above mentioned rights,
+but they do not enforce that you must make derived works or new works
+created by taking bits of source code from the original product
+available under the same license. These type of licenses are generally
+seen as more friendly to commercial enterprises as they will allow
+companies to use bits of code in proprietary software without having to
+release the source code for free. Examples of Permissive licenses are
+Apache, BSD (no matter the number of clauses), and MIT.
+
+### CLA versus DCO
+
+Some Open Source projects are asking contributors to sign up to a
+Contributor License Agreement (CLA) or, alternatively, a Developer
+Certificate of Origin (DCO).
+
+Looking at a number of **CLA**s (in particular Threema, Meta, and
+ImageWorks) it seems that these projects are mainly concerned with the
+following:
+
+-   A person submitting code that they are not the copyright holder for.
+-   Withdrawal of the right to use the submitted code.
+-   The need to apply for patents in future for parts or all of the
+    source code.
+-   A change in the type of license that the source code was given at
+    the time of contribution.
+
+There are also some that touch on the subject of loss and damages which
+may arise from the use of a particular section of code, as well as how a
+request for support will be dealt with in future.
+
+An alternative to a CLA is a Developer Certificate of Origin or DCO. The
+**DCO** was first employed by the Linux Foundation in 2004, and is
+basically a short document that confirms that the person contributing
+code is allowed to do so, gives permission for the project to
+subsequently use it, and is done through the use of a Git commit.
```

Comment:
How does that work, using it through a Git commit? Like, in the commit they write "I agree to these transfer terms?" Or? An example of this might be useful.